packages/shared/components/atoms/tiles/TransactionActivityTile.svelte

```diff
@@ -30,51 +42,70 @@
     export let data: ITransactionActivityData
 
     let asset: IPersistedAsset
+    let asyncStatusTooltipText: string
 
-    $: $selectedAccountAssets, (asset = getAssetFromPersistedAssets(data.assetId))
-    $: isIncomingActivityUnclaimed =
-        (data.direction === ActivityDirection.In || data.isSelfTransaction) &&
-        data.asyncStatus === ActivityAsyncStatus.Unclaimed
-    $: isTimelocked = data.asyncStatus === ActivityAsyncStatus.Timelocked
     $: title = getTitle(data, inclusionState)
     $: subjectLocale = getSubjectLocale(data.subject, data.isShimmerClaiming)
     $: timeDiff = getTimeDiff(data)
+    $: $selectedAccountAssets, (asset = getAssetFromPersistedAssets(data.assetId))
+
+    $: isUnclaimed = data.asyncStatus === ActivityAsyncStatus.Unclaimed
+    $: isTimelocked = data.asyncStatus === ActivityAsyncStatus.Timelocked
+    $: isIncoming = data.direction === ActivityDirection.In
+    $: isIncomingActivityUnclaimed = (isIncoming || data.isSelfTransaction) && isUnclaimed
+    $: isAsyncActivity = isTimelocked || (data.isAsync && (data.direction === ActivityDirection.Out || isUnclaimed))
+    $: {
+        if (isUnclaimed || isTimelocked) {
+            const activityDirectionKey = isIncoming ? 'incoming' : 'outgoing'
+            const activityAsyncStatusKey = isUnclaimed ? 'expirationTime' : 'timelockDate'
+            const textKey = `tooltips.transactionDetails.${activityDirectionKey}.${activityAsyncStatusKey}`
+            asyncStatusTooltipText = localize(textKey)
+        }
+    }
 
     function getTimeDiff(txData: ITransactionActivityData): string {
-        if (isTimelocked) {
-            return getTimeDifference(txData.timelockDate, $time)
-        } else if (txData.isAsync && !txData.isClaimed && txData?.expirationDate) {
-            return getTimeDifference(txData.expirationDate, $time)
-        } else {
-            return localize('general.none')
+        const { asyncStatus, isAsync, isClaimed, expirationDate, timelockDate } = txData
+
+        if (asyncStatus === ActivityAsyncStatus.Timelocked) {
```

Comment:
All values inside this function rely on the function scope instead of global scope, I think that this change could mix contexts and make it a little bit harder to debug/understand, even that I like your suggestion as it is easier to read.